runtime/compiler/runtime/Listener.cpp

```diff
@@ -179,49 +261,64 @@ TR_Listener::serveRemoteCompilationRequests(BaseCompileDispatcher *compiler)
             exit(1);
             }
          }
-      else if (pfd.revents != POLLIN)
-         {
-         fprintf(stderr, "Unexpected event occurred during poll for new connection: revents=%d\n", pfd.revents);
-         exit(1);
-         }
-      do
+      // Check which file descriptor is ready
+      for (size_t fdIndex = 0; fdIndex < numFds; fdIndex++)
          {
-         /* at this stage we should have a valid request for new connection */
-         connfd = accept(sockfd, (struct sockaddr *)&cli_addr, &clilen);
-         if (connfd < 0)
+         if (pfd[fdIndex].revents == 0) // No event on this file descriptor
+            continue;
+         // We have an event and that event can only be POLLIN
+         TR_ASSERT_FATAL(pfd[fdIndex].revents == POLLIN, "Unexpected event occurred during poll for new connection: socketIndex=%zu revents=%d\n", fdIndex, pfd[fdIndex].revents);
+
+         // At this stage we should have a valid request for a new connection
+         pfd[fdIndex].revents = 0; // Reset the event for the next poll operation
+         do
             {
-            if ((EAGAIN != errno) && (EWOULDBLOCK != errno))
+            connfd = accept(pfd[fdIndex].fd, (struct sockaddr *)&cli_addr, &clilen);
+            if (connfd < 0)
                {
-               if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+               if ((EAGAIN != errno) && (EWOULDBLOCK != errno))
                   {
-                  TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "Error accepting connection: errno=%d: %s",
-                                                 errno, strerror(errno));
+                  if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+                     {
+                     TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "Error accepting connection: errno=%d: %s",
+                                                   errno, strerror(errno));
+                     }
                   }
                }
-            }
-         else
-            {
-            struct timeval timeout = { timeoutMs / 1000, (timeoutMs % 1000) * 1000 };
-            if (setsockopt(connfd, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout)) < 0)
+            else // accept() succeeded
                {
-               perror("Can't set option SO_RCVTIMEO on connfd socket");
-               exit(1);
-               }
-            if (setsockopt(connfd, SOL_SOCKET, SO_SNDTIMEO, &timeout, sizeof(timeout)) < 0)
-               {
-               perror("Can't set option SO_SNDTIMEO on connfd socket");
-               exit(1);
+               if (fdIndex == 0) // readiness/liveness probe socket
+                  {
+                  close(connfd);
```

Comment:
Does the probe only need a connection to be accepted and not anything that the jitserver has to send back?

Reply:
Kubernetes documentation says that establishing the connection is enough. No data has to be sent and my tests validated that.